components/console/esp_console.h

```diff
@@ -22,6 +22,7 @@ typedef struct linenoiseCompletions linenoiseCompletions;
 typedef struct {
     size_t max_cmdline_length;  //!< length of command line buffer, in bytes
     size_t max_cmdline_args;    //!< maximum number of command line arguments to parse
+    bool use_psram;             //!< use psram for internal allocations
```

Comment:
What do you think about making `uint32_t alloc_caps //!< heap allocator caps (e.g. MALLOC_CAP_SPIRAM) used  when allocating console objects like cmds, defaults to MALLOC_CAP_DEFAULT` instead? 

This way the API is more flexible and would work even if we were to add some non psram external memory.

This way we can also simply the code (`static s_heap_alloc_caps = MALLOC_CAP_DEFAULT`,  and all the allocs can simply use `s_heap_alloc_caps` when allocating)

What do you think?

Reply:
Yes, I was actually just thinking the same thing. Let me update the PR.

Reply:
updated.

Reply:
updated again